### ISSUE TITLE
Compiled fails on Ubuntu 16.04 server

### ISSUE THREAD

**Issue by ShownX** [tensorflow/tensorflow#6777]
Hi, there,

I compiled the latest v1.0.0-alpha version on a Ubuntu 16.04 server with GPU support. Here is the error I got:
```
/home/xxu/Extern/tensorflow/tensorflow/core/kernels/BUILD:863:1: error while parsing .d file: /home/xxu/.cache/bazel/_bazel_xxu/09a841ffbaae1d24115959abeaae27c9/execroot/tensorflow/bazel-out/local_linux-opt/bin/tensorflow/core/kernels/_objs/gather_functor_gpu/tensorflow/core/kernels/gather_functor_gpu.cu.pic.d (No such file or directory).
In file included from ./tensorflow/core/framework/register_types.h:21:0,
                 from ./tensorflow/core/kernels/gather_functor_gpu.cu.h:23,
                 from tensorflow/core/kernels/gather_functor_gpu.cu.cc:20:
bazel-out/local_linux-opt/genfiles/tensorflow/core/framework/resource_handle.pb.h:12:2: error: #error This file was generated by a newer version of protoc which is
 #error This file was generated by a newer version of protoc which is
  ^
bazel-out/local_linux-opt/genfiles/tensorflow/core/framework/resource_handle.pb.h:13:2: error: #error incompatible with your Protocol Buffer headers. Please update
 #error incompatible with your Protocol Buffer headers.  Please update
  ^
bazel-out/local_linux-opt/genfiles/tensorflow/core/framework/resource_handle.pb.h:14:2: error: #error your headers.
 #error your headers.
  ^
Target //tensorflow/tools/pip_package:build_pip_package failed to build
```

Versions of packages: 
1. The protoc --version is 3.1.0. I installed both C++ and python version;
2. CUDA 7.5, cudnn 5

PS: I compile it without cuda and everything goes well.

**Comment by jart** [comment]
Maybe `@protobuf//:protoc` needs to be added to the `data` attribute of `//tensorflow/tools/pip_package:build_pip_package` and then the script should `export PROTOC=...` to the location of the one built by bazel before calling `bdist_wheel`.

As a workaround, you might want to install https://github.com/google/protobuf/archive/008b5a228b37c054f46ba478ccafa5e855cb16db.tar.gz on your system from source.

What should we do about this @gunan?

**Comment by gunan** [comment]
This is curious.
Maybe there are two different versions of protoc?
This message looks interesting to me
```
This file was generated by a newer version of protoc which is incompatible with your Protocol Buffer headers. Please update your headers.
```

Where did you download protobuf packages you installed?

@jhseu in case he can help.

**Comment by jart** [comment]
@gunan Files like [protobuf_optimized_pip.sh](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/protobuf/protobuf_optimized_pip.sh) lead me to believe that `setup.py bdist_wheel` is doing unholy things as part of the build process, like getting protoc from /usr/bin. Exporting that environment variable in the pip package script would most likely make things a little more hermetically sealed. However it's a shame that Bazel isn't building the pip package. That seems better than delegating to a system build.

**Comment by gunan** [comment]
That is exclusively for ci_build, where we build the cpp optimized protobuf pip package.
That one never gets touched by anything else in our codebase.

Bazel is powerful, it can build a lot of things but IMHO we need to be careful about balancing. 
In a perfect world we let bazel build everything it can.
In real life, the more we let bazel build things, the more painful our developer experience is going to be as most developers use 2-core/4-core machines. Together with bazel bugs, building tensorflow from sources becomes more and more painful with more dependencies.

**Comment by ShownX** [comment]
@gunan @jart Thanks your response!

The server indeed has multiple version of protobuf because some users run Caffe which need protobuf v2.5. I tested the libprotobuf header in ```/usr/local/include``` and version of protoc. Both of them are v3.1.0. It is downloaded and compiled from release page [https://github.com/google/protobuf/releases/tag/v3.1.0](https://github.com/google/protobuf/releases/tag/v3.1.0).

Finally I successfully compiled the latest tensorflow. And I want to provide some information for the tensorflow team:
I download CUDA 8 and recompile from source, it works. It seems it is problem with CUDA 7.5 and CUDNN 5.1.3. Therefore, the very wierd information provided is wrong.
